### PR TITLE
Added Config.js file and added the env variables

### PR DESCRIPTION
**Pull request by alenjohn005**

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "dialect": "postgres"
+  },
+  "production": {
+    "username": "pc22",
+    "password": "Lilac@PC22#",
+    "database": "mealmedatabase",
+    "host": "localhost",
+    "dialect": "postgres"
+  }
+}
```
Comment (biplobmanna) [inline config/config.json]: Good to see this config. This is the standard way of setting up any environment. 
It was missing in the code sent by @Sharooq-KN 
But now that it's here, it's good.

File: db/db.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const { Sequelize } = require("sequelize");
+const sequelize = new Sequelize("sqlite::memory:");
+module.exports = sequelize;
+
+const sequelize = new Sequelize({
+  dialect: "postgres",
+  database: process.env.DATABASE,
+  username: process.env.DATABASE_USER,
+  password: process.env.DATABASE_PASSWORD,
+  host: "localhost",
```
Comment (biplobmanna) [inline db/db.js:10]: This is also a good implementation of defining the object. 
The `host` directive will be useful when deploying in containers.

File: migrations/202202151654-create-user.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    await queryInterface.createTable("UserType", {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER,
+      },
+      Type: {
+        defaultValue: "Customer",
+        type: Sequelize.ENUM("Staff(admin & kitchen)", "Customer", "Driver"),
```
Comment (biplobmanna) [inline migrations/202202151654-create-user.js]: Good use of enum. 
Naming could be better. 
It's better to not use any special characters in the naming as well.

Comment (alenjohn005) [reply migrations/202202151654-create-user.js]: OK, I shall remove the ampersand

File: config/config.json (hunk 1 above)
Comment (biplobmanna) [reply config/config.json]: But, the better usage is to convert this file `config.json` from **JSON** to **JS**

And, export the entire object.

Comment (alenjohn005) [reply config/config.json]: ok

Comment (biplobmanna) [reply config/config.json]: Also, after converting to **JS** populate the configuration directives using the ENV variables. 

For eg:
```javascript
"username": process.env.DB_USER
```

File: models/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require(__dirname + '/../config/config.json')[env];
+const db = {};
+
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
```
Comment (biplobmanna) [inline models/index.js]: If the config file was converted from **JSON** to **JS** and the directives populated using the ENV_VARS

then this configuration will be more modular and work across multiple environments including local, prod, docker, kubernetes, etc...

Comment (biplobmanna) [reply models/index.js]: Also remember, that using **sequilize-cli** for generating models creates plural table names by default. Be careful while using it.

File: migrations/202202151654-create-user.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          allowNull: false,
+        },
+      },
+    });
+  },
+  down: async (queryInterface, Sequelize) => {
+    // we can do this because it is the first migration
+    await queryInterface.dropAllTables();
+  },
+};
```
Comment (biplobmanna) [inline migrations/202202151654-create-user.js]: Also, where is the migration for user-type ?

**Review by biplobmanna — CHANGES_REQUESTED** [review]
Read through the comments and make changes where necessary.

File: models/index.js (hunk 4 above)
Comment (alenjohn005) [reply models/index.js]: I have tested and it creates the required types

**Review by biplobmanna — APPROVED** [review]
Looks good. PR approved.

**Pull request merged**